src/Compilers/CSharp/Portable/Symbols/Source/LambdaSymbol.cs

```diff
@@ -278,15 +278,37 @@ internal override OneOrMany<SyntaxList<AttributeListSyntax>> GetAttributeDeclara
                 default;
         }
 
-        internal void GetDeclarationDiagnostics(BindingDiagnosticBag addTo)
+        internal void GetDeclarationDiagnostics(out ImmutableArray<BoundAttribute> boundAttributes, out ImmutableArray<BoundAttribute> returnBoundAttributes, BindingDiagnosticBag addTo)
         {
             foreach (var parameter in _parameters)
             {
                 parameter.ForceComplete(locationOpt: null, cancellationToken: default);
             }
 
-            GetAttributes();
-            GetReturnTypeAttributes();
+            boundAttributes = default;
+            returnBoundAttributes = default;
+
+            var bag = _lazyCustomAttributesBag;
+            if (bag == null || bag.IsSealed)
+            {
+                LoadAndValidateAttributes(
+                    GetAttributeDeclarations(),
+                    ref _lazyCustomAttributesBag,
+                    boundAttributes: out boundAttributes,
+                    AttributeLocation.None,
+                    binderOpt: OuterBinder);
+            }
+
+            var returnBag = _lazyReturnTypeCustomAttributesBag;
+            if (returnBag == null || returnBag.IsSealed)
+            {
+                LoadAndValidateAttributes(
+                    GetReturnTypeAttributeDeclarations(),
+                    ref _lazyReturnTypeCustomAttributesBag,
+                    boundAttributes: out returnBoundAttributes,
+                    AttributeLocation.Return,
+                    binderOpt: OuterBinder);
+            }
```

Comment:
I think you might not like this. Let me know and I'll try to refactor it so that `GetAttributes` and `GetReturnTypeAttributes` have the needed out parameters.